### PR TITLE
[Feature] Support analyze hive table

### PR DESCRIPTION
**Pull request by Youngwb**
#23327
query hive table statistics of table level

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 63 / 72 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|1|2|50.00%|[116]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|10|16|62.50%|[298, 299, 300, 312, 313, 314]|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|4|5|80.00%|[138]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|11|12|91.67%|[68]|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|28|28|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticStorage.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/statistic_result_writer.cpp|0|1|00.00%|[156]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 29 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/statistic_result_writer.cpp|0|29|00.00%|[156, 157, 406, 408, 411, 413, 414, 415, 416, 417, 418, 419, 421, 422, 424, 425, 426, 427, 428, 429, 430, 431, 432, 435, 436, 438, 439, 440, 442]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 135 / 164 (82.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|0|3|00.00%|[116, 117, 118]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|3|5|60.00%|[182, 183]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTableColumnKey.java|10|13|76.92%|[32, 35, 42]|
|:large_blue_circle:|com/starrocks/connector/ConnectorColumnStatsCacheLoader.java|43|55|78.18%|[63, 65, 66, 67, 68, 104, 105, 114, 118, 123, 131, 149]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|9|11|81.82%|[62, 247]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|31|36|86.11%|[144, 169, 176, 177, 178]|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|9|10|90.00%|[143]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|17|18|94.44%|[365]|
|:large_blue_circle:|com/starrocks/connector/ConnectorTableColumnStats.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticStorage.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 29 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/runtime/statistic_result_writer.cpp|0|29|00.00%|[156, 157, 406, 408, 411, 413, 414, 415, 416, 417, 418, 419, 421, 422, 424, 425, 426, 427, 428, 429, 430, 431, 432, 435, 436, 438, 439, 440, 442]|

**Review by gengjun-git — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java [hunk 1]
```diff
@@ -60,6 +63,13 @@ public class CachedStatisticStorage implements StatisticStorage {
             .executor(statsCacheRefresherExecutor)
             .buildAsync(new ColumnBasicStatsCacheLoader());
 
+    AsyncLoadingCache<ConnectorTableColumnKey, Optional<ConnectorTableColumnStats>> connectorTableCachedStatistics =
+            Caffeine.newBuilder().expireAfterWrite(Config.statistic_update_interval_sec * 2, TimeUnit.SECONDS)
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java:67]: why use this value

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java [hunk 2]
```diff
@@ -277,15 +281,39 @@ public List<String> listPartitionNamesByValue(String catalogName, String dbName,
...
     public Statistics getTableStatistics(OptimizerContext session,
                                          String catalogName,
                                          Table table,
                                          Map<ColumnRefOperator, Column> columns,
                                          List<PartitionKey> partitionKeys,
                                          ScalarOperator predicate) {
-        Optional<ConnectorMetadata> connectorMetadata = getOptionalMetadata(catalogName);
-        return connectorMetadata.map(metadata ->
-                metadata.getTableStatistics(session, table, columns, partitionKeys, predicate)).orElse(null);
+        Statistics statistics = getTableStatisticsFromInternalStatistics(table, columns);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:309]: If there is unknown, could we use Internal estimated columns + connector column ？

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java:67]: It follows the cache of olap table

**Review by stephen-shelby — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java (hunk 2 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java:309]: I am worried that the inconsistent collection methods will lead to worse plans

**Comment by kShines** [comment]
Can the exterior also support the hll algorithm, for example hive should not support hll, how to calculate ndv in this case

**Pull request merged**